assignments/assignment_5_md_version.md

```diff
@@ -0,0 +1,89 @@
+```python
+# This is just for you so that you can see where the numbers are coming from
+np.random.seed(12345)
+
+cars = np.random.randint(500,8000,size=500)
+np.save('./cars.npy',cars)
+
+value = np.random.uniform(size=500)
+np.save('./value.npy',value)
+```
+
+# Assignment 5
+
+Dan a university student wants to buy a car. He knows the price of each he is
+considering and he assigned a value per pound ratio to each of the cars (based
+on horsepower, consumption etc.). He saved this data in two arrays (clearly, why
+wouldn't he?), where the order of the cars was the same. Obviously Dan doesn't
+want a slow car, he wants the car to have at least $100$ horsepower, which start
+from $1500$ pounds. So Dan did the following:
+
+```python
+# Import numpy
+import numpy as np
+```
+
+First he loaded the arrays he created earlier, in the variable called cars and
+value using a numpy function
+[np.load()](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.load.html).
+
+```python
+cars = np.load('./cars.npy')
+value = np.load('./value.npy')
+```
+
+Now he choose from the array the cars, which cost more, than the lower bound he
+set for himself.
+
+```python
+cool_cars = cars[cars>1500]
```

Comment:
They haven't come across the idea of indexing yet either.  

So, they don't yet have the idea you can get an element from a sequence with something like `sequence[index]`.  Here the indexing is more complicated.

They first have to understand something the element-wise nature of comparisons, on arrays:

```
In [2]: a = np.arange(10)                                                                                                          

In [3]: a                                                                                                                          
Out[3]: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [4]: a > 5                                                                                                                      
Out[4]: 
array([False, False, False, False, False, False,  True,  True,  True,
        True])
```

After that, they need the idea that they can use these True / False values to index into the array.

I don't think we will cover that on Friday - it's a bit more advanced.